packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,993 @@
+    const globalSecondaryIndexes = this.configureReplicaGlobalSecondaryIndexes(props.globalSecondaryIndexOptions);
+    const pointInTimeRecovery = props.pointInTimeRecovery ?? this.tableOptions.pointInTimeRecovery;
+    const contributorInsights = props.contributorInsights ?? this.tableOptions.contributorInsights;
+    const readProvisionedThroughputSettings = props.readCapacity
+      ? this.configureReadProvisioning(props.readCapacity)
+      : this.readProvisioning;
+
+    return {
+      region: props.region,
+      readProvisionedThroughputSettings,
+      globalSecondaryIndexes,
+      deletionProtectionEnabled: props.deletionProtection ?? this.tableOptions.deletionProtection,
+      tableClass: props.tableClass ?? this.tableOptions.tableClass,
+      sseSpecification: this.configureReplicaSseSpecification(props.region),
+      kinesisStreamSpecification: props.kinesisStream
+        ? { streamArn: props.kinesisStream.streamArn }
+        : undefined,
+      contributorInsightsSpecification: contributorInsights !== undefined
+        ? { enabled: contributorInsights }
+        : undefined,
+      pointInTimeRecoverySpecification: pointInTimeRecovery !== undefined
+        ? { pointInTimeRecoveryEnabled: pointInTimeRecovery }
+        : undefined,
+    };
+  }
+
+  private configureSseSpecification(encryption: TableEncryptionV2): CfnGlobalTable.SSESpecificationProperty {
+    if (encryption.type === TableEncryption.DEFAULT) {
+      return { sseEnabled: false };
+    }
+
+    return { sseEnabled: true, sseType: 'KMS' };
+  }
+
+  private configureReplicaSseSpecification(region: string): CfnGlobalTable.ReplicaSSESpecificationProperty | undefined {
+    if (!this.encryption || this.encryption.type !== TableEncryption.CUSTOMER_MANAGED) {
+      return undefined;
+    }
+
+    if (region === this.region) {
+      return { kmsMasterKeyId: this.encryption.tableKey.keyArn };
+    }
+
+    if (this.encryption && this.encryption.type === TableEncryption.CUSTOMER_MANAGED) {
+      const regionInReplicaKeyArns = this.encryption.replicaKeyArns.hasOwnProperty(region);
+      if (!regionInReplicaKeyArns) {
+        throw new Error(`You must specify a KMS key ARN for each replica table when encryption type is ${TableEncryption.CUSTOMER_MANAGED}`);
+      }
+    }
+
+    return { kmsMasterKeyId: this.encryption.replicaKeyArns[region] };
+  }
+
+  private configureReadProvisioning(readCapacity: Capacity): CfnGlobalTable.ReadProvisionedThroughputSettingsProperty {
+    if (readCapacity.mode === CapacityMode.FIXED) {
+      return { readCapacityUnits: readCapacity.units };
+    }
+
+    return {
+      readCapacityAutoScalingSettings: {
+        minCapacity: readCapacity.minCapacity,
+        maxCapacity: readCapacity.maxCapacity,
+        targetTrackingScalingPolicyConfiguration: {
+          targetValue: readCapacity.targetUtilizationPercent,
+        },
+      },
+    };
+  }
```

Comment:
Same for these, can we not defer to the implementations?

Make sure they return `any` in the declaration, but they can use TypeScript `satisfies` so you get type checking:

```ts
abstract class Capacity {
  public static onDemand(): Capacity {
    return new class extends Capacity {
      public renderCapacity(): any { return undefined; } /* Note return type declaration, is necessary for jsii */
    }();
  }

  public static autoScaled(/* ... */): Capacity {
    return new class extends Capacity {
      public renderCapacity(): any {
       return {
        readCapacityAutoScalingSettings: { minCapacity: /* ... */ },
       } satisifies CfnGlobalTable.ReadProvisionedThroughputSettingsProperty;
    }
  }

  abstract public renderCapacity(): any;
}
```